src/org/parosproxy/paros/core/scanner/VariantMultipartFormParameters.java

```diff
@@ -0,0 +1,195 @@
+			// http://www.regexplanet.com/share/index.html?share=yyyyyythear (Click Java, set case 
+			// insensitive & DOTALL, and hit "test")
+
+	private List<NameValuePair> params = Collections.emptyList();
+	private List<MultipartFormParameter> multiPartParams = new ArrayList<MultipartFormParameter>();
+
+	@Override
+	public void setMessage(HttpMessage msg) {
+		if (msg == null) {
+			throw new IllegalArgumentException("Parameter message must not be null.");
+		}
+
+		String contentType = msg.getRequestHeader().getHeader(HttpHeader.CONTENT_TYPE);
+		if (contentType == null || !contentType.toLowerCase().startsWith("multipart/form-data")) {
+			return;
+		}
+
+		ArrayList<NameValuePair> extractedParameters = new ArrayList<>();
+		int position = 0;
+		int offset = 0;
+		String boundary = getBoundary(contentType) + HttpHeader.CRLF;
+		for (String part : msg.getRequestBody().toString().split(boundary)) {
+			if (!StringUtils.isBlank(part)) {
+				String partHeaderLine = part.substring(0, part.indexOf(HttpHeader.CRLF + HttpHeader.CRLF));
+				part = boundary + part;
+				Matcher nameMatcher = FIELD_NAME_PATTERN.matcher(partHeaderLine);
+				Matcher valueMatcher = FIELD_VALUE_PATTERN.matcher(part);
+				nameMatcher.find();
+				valueMatcher.find();
+				if (StringUtils.isBlank(valueMatcher.group("value"))) {
+					valueMatcher.find();
+					// Need to skip one find for some reason...
+					// https://regex101.com/r/4ig6Wk/1
+					// http://fiddle.re/23cudd (Click Java, hit "test")
+				}
+
+				String name = nameMatcher.group("name");
+				// Value doesn't include boundary, headerline, or double CRLF
+				String value = part.replaceAll(boundary + partHeaderLine + HttpHeader.CRLF + HttpHeader.CRLF, "");
+				value = value.replaceAll(getBoundary(contentType) + "--" + HttpHeader.CRLF, "");// Strip final boundary
+				extractedParameters.add(new NameValuePair(NameValuePair.TYPE_POST_DATA, name, value, position));
+				int start = offset + part.indexOf(HttpHeader.CRLF + HttpHeader.CRLF) + 4; // 4 for two CRLFs
+				int end = offset + part.length();
+				if (LOGGER.isDebugEnabled()) {
+					LOGGER.debug("Name: " + name + " O: " + offset + " S: " + start + " E: " + end + " Pos: " + position);
+				}
+				multiPartParams.add(new MultipartFormParameter(name, valueMatcher.group("value"), start, end, position,
+						MultipartFormParameter.Type.GENERAL));
+				if (LOGGER.isDebugEnabled()) {
+					LOGGER.debug("Group name: " + name + " value: " + valueMatcher.group("value"));
+				}
+				if (partHeaderLine.contains("filename=")) {
+					// Extract the filename
+					Matcher fnValueMatcher = FILENAME_PART_PATTERN.matcher(part);
+					fnValueMatcher.find();
+					String fnValue = fnValueMatcher.group("filename");
+					extractedParameters.add(new NameValuePair(NameValuePair.TYPE_POST_DATA, name, fnValue, ++position));
+					int fnStart = offset + part.indexOf(fnValue);
+					int fnEnd = offset + (fnStart + fnValue.length());

```

Comment:
Should be just `fnStart + fnValue.length()` (`fnStart` already has the `offset`).